src/tree/hist/histogram.h

```diff
@@ -76,6 +84,20 @@ class HistogramBuilder {
       buffer_.Reset(this->n_threads_, n_nodes, space, target_hists);
     }
 
+    constexpr float kColsampleTh = 0.1;
+    bool column_sampling = (column_sampler_ != nullptr) &&
```

Comment:
When column sampler is nullptr?

Reply:
I set it to nullptr in [tests ](https://github.com/dmlc/xgboost/pull/8319/files#diff-4bdebd351f1f1f377c74e6764eaf56ff9c6e1b5f9a123ab51665003f164b56d2R49)of the histogram builder without using this optimization.